parser/charset/encoding_base.go

```diff
@@ -128,3 +128,16 @@ func Slice(s string) (b []byte) {
 	pBytes.Cap = pString.Len
 	return
 }
+
+// String converts slice to string without copy.
+// Use it at your own risk.
+func String(b []byte) (s string) {
```

Comment:
```suggestion
func HackString(b []byte) (s string) {
```

Could you prepend "Hack" to `String` and `Slice` to make it more obvious?

Reply:
Maybe we can move String() and Slice() to other file(like `util`) later.
And delete `hack.Slice` in `tidb/util/hack/hack.go`